pytensor/tensor/nlinalg.py

```diff
@@ -231,6 +231,39 @@ def __str__(self):
 det = Det()
 
 
+class SLogDet(Op):
+    """
+    Compute the log determinant and its sign of the matrix. Input should be a square matrix.
+    """
+
+    __props__ = ()
+
+    def make_node(self, x):
+        x = as_tensor_variable(x)
+        assert x.ndim == 2
+        sign = scalar(dtype=x.dtype)
```

Comment:
Shouldn't sign be an integer?

Reply:
I saw that `dtype` of the sign return value in numpy depends on `dtype` of the input matrix:

```python
import numpy as np
x = np.array([[0, 5], [3, 1]])

sign, _ = np.linalg.slogdet(x.astype('double'))
print(sign.dtype)  # dtype('float64')

sign, _ = np.linalg.slogdet(x.astype('single'))
print(sign.dtype)  # dtype('float32')
```
That's why I set it with `x`'s dtype. Should I change it to `int`?

Reply:
Ah no, then leave it as is!